pkg/authz/permissions.go

```diff
@@ -79,3 +93,93 @@ func checkAccess(ctx context.Context, c *kubernetes.Clientset, np NamespacedPerm
 
 	return r.Status.Allowed, nil
 }
+
+func creatUnwantedPermissionsList(ctx context.Context, cfg *rest.Config, unwantedPermissions []string, ns string, definedRules map[[4]string]int) ([]string, error) {
+	c, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return []string{}, fmt.Errorf("error creating the client: %w", err)
+	}
+
+	ssrr := authzv1.SelfSubjectRulesReview{
+		ObjectMeta: metav1.ObjectMeta{Namespace: ns},
+		Spec:       authzv1.SelfSubjectRulesReviewSpec{Namespace: ns},
+	}
+
+	r, err := c.AuthorizationV1().SelfSubjectRulesReviews().Create(ctx, &ssrr, metav1.CreateOptions{})
+	if err != nil {
+		return []string{}, err
+	}
+	resourceRules := r.Status.ResourceRules
+	for _, rr := range resourceRules {
+		APIGroups := rr.APIGroups
+		if len(APIGroups) == 0 {
+			APIGroups = []string{""}
+		}
+		ResourceNames := rr.ResourceNames
+		if len(ResourceNames) == 0 {
+			ResourceNames = []string{""}
+		}
+		for _, ag := range APIGroups {
+			for _, rs := range rr.Resources {
+				for _, rn := range ResourceNames {
```

Comment:
If `ResourceNames` is empty we are not executing the next loops right?

Reply:
See the line no. 119 to 121. I am adding an empty string to the slice.